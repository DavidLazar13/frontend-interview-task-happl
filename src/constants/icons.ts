import Logo from '@/assets/logo.svg'
import Home from '@/assets/home.svg'

export const ICONS = {
  logo: Logo,
  home: Home,
}
