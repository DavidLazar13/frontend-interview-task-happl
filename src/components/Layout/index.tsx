import React from 'react'
import Link from 'next/link'
import { INDEX } from '@/constants/routes'
import { useRouter } from 'next/router'
import Icon from '@/components/Icon'

type Props = {
  children: React.ReactElement
}

const Layout = ({ children }: Props) => {
  const router = useRouter()

  return (
    <>
      <div className="bg-green-400 flex fixed items-center md:items-stretch md:flex-col md:w-72 h-16 md:h-screen md:inset-y-0 inset-x-0 md:space-y-4 space-x-3 md:space-x-0 px-3 py-3 md:py-4">
        <div className="flex items-center space-x-2">
          <div className="rounded-md overflow-hidden">
            <Icon icon="logo" width={33} height={33} />
          </div>
          <p className="hidden md:flex text-sm text-white">Happl</p>
        </div>
        <nav>
          <Link
            href={INDEX}
            className={`flex items-center h-9 px-2 rounded-md text-sm text-white hover:bg-green-500 focus:outline-none focus:ring-2 focus:ring-green-300 ${
              router.pathname === INDEX ? 'bg-green-500' : ''
            }`}
          >
            <span className="mr-2">
              <Icon icon="home" />
            </span>
            Home
          </Link>
        </nav>
      </div>
      <div className="md:pl-72 pt-16 md:pt-0 flex flex-col flex-1">
        <main className="flex-1">{children}</main>
      </div>
    </>
  )
}

export default Layout
