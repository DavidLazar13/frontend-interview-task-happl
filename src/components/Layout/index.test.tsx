import { render, fireEvent } from '@testing-library/react'
import Layout from '@/components/Layout'
import '@testing-library/jest-dom'
import { INDEX } from '@/constants/routes'
import mockRouter from 'next-router-mock'
import router from 'next/router'

const children = 'Children'

describe('Layout', () => {
  beforeEach(() => {
    mockRouter.setCurrentUrl(`${INDEX}#`)
  })

  it('renders as expected', () => {
    const { container } = render(
      <Layout>
        <p>{children}</p>
      </Layout>
    )
    expect(container).toMatchSnapshot()
  })

  it('renders logo as expected', () => {
    const { getByTestId } = render(
      <Layout>
        <p>{children}</p>
      </Layout>
    )
    expect(getByTestId('icon-logo')).toBeInTheDocument()
  })

  it('renders name as expected', () => {
    const { getByText } = render(
      <Layout>
        <p>{children}</p>
      </Layout>
    )
    expect(getByText('Happl')).toBeInTheDocument()
  })

  it('renders children as expected', () => {
    const { getByText } = render(
      <Layout>
        <p>{children}</p>
      </Layout>
    )
    expect(getByText(children)).toBeInTheDocument()
  })

  it('renders home as expected', () => {
    const { getByText, getByTestId } = render(
      <Layout>
        <p>{children}</p>
      </Layout>
    )
    expect(getByText('Home')).toBeInTheDocument()
    expect(getByText('Home')).toHaveAttribute('href', INDEX)
    expect(getByTestId('icon-home')).toBeInTheDocument()
  })

  it('makes expected calls on home  click', async () => {
    const { getByText } = render(
      <Layout>
        <p>{children}</p>
      </Layout>
    )
    fireEvent.click(getByText('Home'))
    expect(router.asPath).toBe(INDEX)
  })
})
