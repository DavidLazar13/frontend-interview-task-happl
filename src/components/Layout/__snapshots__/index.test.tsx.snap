// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout renders as expected 1`] = `
<div>
  <div
    class="bg-green-400 flex fixed items-center md:items-stretch md:flex-col md:w-72 h-16 md:h-screen md:inset-y-0 inset-x-0 md:space-y-4 space-x-3 md:space-x-0 px-3 py-3 md:py-4"
  >
    <div
      class="flex items-center space-x-2"
    >
      <div
        class="rounded-md overflow-hidden"
      >
        <svgr-url
          data-testid="icon-logo"
          height="33"
          strokewidth="1.5"
          width="33"
        />
      </div>
      <p
        class="hidden md:flex text-sm text-white"
      >
        Happl
      </p>
    </div>
    <nav>
      <a
        class="flex items-center h-9 px-2 rounded-md text-sm text-white hover:bg-green-500 focus:outline-none focus:ring-2 focus:ring-green-300 bg-green-500"
        href="/"
      >
        <span
          class="mr-2"
        >
          <svgr-url
            data-testid="icon-home"
            height="20"
            strokewidth="1.5"
            width="20"
          />
        </span>
        Home
      </a>
    </nav>
  </div>
  <div
    class="md:pl-72 pt-16 md:pt-0 flex flex-col flex-1"
  >
    <main
      class="flex-1"
    >
      <p>
        Children
      </p>
    </main>
  </div>
</div>
`;
